ui/app/components/namespace-picker.js

```diff
@@ -63,20 +66,52 @@ export default class NamespacePicker extends Component {
      *   | 'parent' | 'parent'       | 'parent'       |
      *   | 'child'  | 'parent/child' | 'parent/child' |
      */
-    return [
-      // TODO: Some users (including HDS Admin User) should never see the root namespace. Address this in a followup PR.
-      { id: 'root', path: '', label: 'root' },
+    const options = [
       ...(namespace?.accessibleNamespaces || []).map((ns) => {
         const parts = ns.split('/');
         return { id: parts[parts.length - 1], path: ns, label: ns };
       }),
     ];
+
+    // Conditionally add the root namespace
+    if (this.auth.authData.userRootNamespace === '') {
+      options.unshift({ id: 'root', path: '', label: 'root' });
+    }
+
+    return options;
   }
 
   get namespaceLabel() {
     return this.searchInput === '' ? 'All namespaces' : 'Matching namespaces';
   }
 
+  async resetList() {
+    this.allNamespaces = this.#getOptions(this.namespace);
+    this.selected = this.#getSelected(this.allNamespaces, this.namespace);
+
+    // default options and namespace count
+    this.options = this.allNamespaces;
+    this.namespaceCount = this.options.length;
+  }
+
+  @action
+  onSearchInput(event) {
+    this.searchInput = event.target.value;
+
+    // Filter options based on the search input
+    if (this.searchInput.trim() === '') {
+      // If the search input is empty, reset to all namespaces
+      this.options = this.allNamespaces;
+    } else {
+      // Filter namespaces based on the search input
+      this.options = this.allNamespaces.filter((option) =>
+        option.label.toLowerCase().includes(this.searchInput.toLowerCase())
+      );
+    }
+
+    this.namespaceCount = this.options.length;
```

Comment:
I believe `this.options` and `this.namespaceCount` could benefit from dynamically updating by being getters as well. That way we don't have to manage another tracked state 🤔 
```js
get namespaceOptions() {
   if (this.searchInput.trim() === '') {
     // If the search input is empty, reset to all namespaces
     return this.allNamespaces;
   } else {
      // Filter namespaces based on the search input
      return this.allNamespaces.filter((option) =>
        option.label.toLowerCase().includes(this.searchInput.toLowerCase())
   );
   }
}
```

Reply:
This is SO much better and cleaner, thank you!! ✨